### PR TITLE
[CSM O11Y] Fix issue when CSM optional labels are present in server metrics

### PR DESCRIPTION
**Pull request by yijiem**

### CODE REVIEW

File: src/cpp/ext/otel/otel_plugin.h [hunk 1]
```diff
@@ -66,6 +66,15 @@ class LabelsIterable {
 // through the OpenTelemetry plugin.
 class LabelsInjector {
  public:
+  // Some optional labels may only be set on client or server. We use this enum
+  // to specify the entity of the caller for the APIs that need this
+  // information.
+  enum class Entity {
```
Comment (yashykt) [inline src/cpp/ext/otel/otel_plugin.h:72]: Throughout gRPC Core, we've just been using a simple `is_client` bool for this. If we were to do this, I would prefer doing it all over.

Comment (markdroth) [reply src/cpp/ext/otel/otel_plugin.h:72]: +1 -- let's just use a bool here.  We're not expecting to have any context other than client or server in the future, so there's no need to leave room for future options.

**Comment by markdroth** [comment]
I assume this will need to be back-ported to 1.61 before the release goes out.

File: src/cpp/ext/otel/otel_plugin.h (hunk 1 above)
Comment (yijiem) [reply src/cpp/ext/otel/otel_plugin.h]: Done.

**Review by yashykt — APPROVED** [review]

File: src/cpp/ext/csm/metadata_exchange.cc [hunk 2]
```diff
@@ -493,9 +498,13 @@ bool ServiceMeshLabelsInjector::AddOptionalLabels(
 }
 
 size_t ServiceMeshLabelsInjector::GetOptionalLabelsSize(
+    bool is_client,
     absl::Span<const std::shared_ptr<std::map<std::string, std::string>>>)
     const {
-  return 2;
+  if (is_client) {
```
Comment (markdroth) [inline src/cpp/ext/csm/metadata_exchange.cc:504]: Can write this as `return is_client ? 2 : 0`.

**Review by markdroth — APPROVED** [review]

File: src/cpp/ext/csm/metadata_exchange.cc (hunk 2 above)
Comment (yijiem) [reply src/cpp/ext/csm/metadata_exchange.cc]: Done.

**Review by markdroth — APPROVED** [review]

**Review by markdroth — APPROVED** [review]

**Pull request closed without merging**